src/components/engagement/engagement.service.ts

```diff
@@ -122,6 +122,14 @@ export class EngagementService {
     @Logger(`engagement.service`) private readonly logger: ILogger
   ) {}
 
+  async setEngagementIdsIntoStatusObjects(engagements: EngagementListOutput) {
```

Comment:
You can just move this into the list function or at least make this private & not async.